### DISCUSSION TITLE
Add Uno-Check after-build actions/capabilitys

### DISCUSSION THREAD

**Discussion by DevTKSS** [unoplatform/uno#18740] [Ideas]
It would be a really great feature if Uno-Check could be capable of some CLI commands to help not only at generating a new uno project, but also could do some in-project management

### In the solution or csproj root folder running Uno-Check could do:

First of all, it would determine the actual application we are working on like WinUI template, .Net 6 ... whatever

#### Target Management:

`Uno-Check add Target -windows | -android | -ios | -wasm`
`Uno-Check remove Target ...`

Steps to do at this moment: https://platform.uno/docs/articles/guides/how-to-add-platforms-existing-project.html
 go to github.com/unoplatform/uno.templates/...
and get the needed template we want to add:
- folder with substructure and files like programm.cs we would copy from that temp project described in the documentation link above
- lookup our csproj file and add the desired target in the TargetFramework tag

#### File Management

`Uno-Check add File -Type <Model | ViewModel (in Mvvm Pattern usage) | Page | ContentDialog | Ressource | ... > SomeName --Markup (optional) <Xaml | csharp>

add files from uno.template, which is done manually via File Manager in the specific IDE: `Add>New Item`

**Comment by jeromelaban** [comment]
Thanks for the suggestion. uno-check is not a tool intended to do project manipulations, it's only there to validate the environment's configuration. That being said, `dotnet new` could be enhanced to include this somehow, or directly from IDE's commands.

**Reply by DevTKSS** [reply]
Never done anything with dotnet  before, so I dont know its capabilitys on that to be honest. It was just a idea because this cli behaviour is e.g. build in in Angular-CLI which is doing just that service from dev terminal.
Is there a way the dotnet cli could be influenced as you said or what does that mean?
Also discovered that there is actually a NewItem Folder in the Visual Studio folder system that says xaml which made me think, what might be the argument to not put the xaml and xaml.cs in there? Could you explain that to me?

**Reply by jeromelaban** [reply]
At this time, the source for the VS integration is not public, but we already provide item templates for pages, controls, and alike. We're not yet providing items for creating missing platforms.

`dotnet new` sources are available in the https://github.com/unoplatform/uno.templates repository, where it's possible to add new templates.
